plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3AsyncService.java

```diff
@@ -243,8 +257,12 @@ static ClientOverrideConfiguration buildOverrideConfiguration(final S3ClientSett
                 )
                 .build()
         );
+        ClientOverrideConfiguration.Builder builder = ClientOverrideConfiguration.builder();
+        if (scheduledExecutorService != null) {
+            builder = builder.scheduledExecutorService(scheduledExecutorService);
+        }
 
-        return ClientOverrideConfiguration.builder()
+        return builder.scheduledExecutorService(scheduledExecutorService)
```

Comment:
You conditionally call `.scheduledExecutorService(scheduledExecutorService)` above. I think this should not be here?

Reply:
indeed, thanks for catching this, fixing